### ISSUE TITLE
Why AMP make backward speed more slow?

### ISSUE THREAD

**Issue by jinfagang** [pytorch/pytorch#71301]
### 🐛 Describe the bug

Here is my test code:

```python
import cv2
import numpy as np
import torch
import torch.nn.functional as F
from torchvision.models.resnet import resnet50
from torchvision.datasets.fakedata import FakeData
from torchvision import transforms
import time


# ----------------------------- test on half 
t0 = time.perf_counter()


transform = transforms.Compose(
    [
        transforms.Resize(224),
        transforms.ToTensor(),
    ]
)
model = resnet50().cuda()
optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
scaler = torch.cuda.amp.GradScaler()
data = FakeData(size=100, transform=transform)
ds = torch.utils.data.DataLoader(data, batch_size=3)
for epoch in range(10):
    for input, target in ds:
        optimizer.zero_grad()
        with torch.cuda.amp.autocast():
            input = input.cuda()
            target = target.cuda()
            t00 = time.perf_counter()
            output = model(input)
            loss = F.cross_entropy(output, target)
            t01 = time.perf_counter()
            print('model forward： ', t01 - t00)
            print(loss.dtype)
        t01 = time.perf_counter()
        scaler.scale(loss).backward()
        scaler.step(optimizer)
        scaler.update()
        t02 = time.perf_counter()
        print('loss backward： ', t02 - t01)

# --------------- without AMP
# model = resnet50().cuda()
# optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
# data = FakeData(size=100, transform=transform)
# ds = torch.utils.data.DataLoader(data, batch_size=3)
# for epoch in range(10):
#     for input, target in ds:
#         optimizer.zero_grad()
#         input = input.cuda()
#         target = target.cuda()
#         t00 = time.perf_counter()
#         output = model(input)
#         loss = F.cross_entropy(output, target)
#         print(loss.dtype)
#         t01 = time.perf_counter()
#         print('model forward： ', t01 - t00)
#         loss.backward()
#         optimizer.step()
#         t02 = time.perf_counter()
#         print('loss backward： ', t02 - t01)

t1 = time.perf_counter()
print('t: ', t1 - t0)

```

Result:

```
using AMP:
loss backward：  0.03126340499147773
model forward：  0.01168522983789444

without AMP:
loss backward：  0.017987865954637527
model forward：  0.020431655924767256

```

As you can see, if fp16 enable, the forward might faster, but backward much more slower , why? My test GPU was T4

### Versions

pytorch 1.10

**Comment by ngimel** [comment]
You are not doing necessary synchronizations when measuring time.

**Comment by jinfagang** [comment]
@ngimel which part need synchronization?

**Comment by ngimel** [comment]
You'd need to put `torch.cuda.synchronize()` before every `time.perf_counter()`,  
You could also  use Timer https://pytorch.org/docs/stable/benchmark_utils.html?highlight=timer#torch.utils.benchmark.Timer, but it's a bit tricky to time backward with `Timer` because it runs the snippet multiple times.

**Comment by Windaway** [comment]
Speedtest
![image](https://user-images.githubusercontent.com/4530735/213174869-88a11f32-c431-4cc1-ba77-343ca1df7a4c.png)
NCHW
![image](https://user-images.githubusercontent.com/4530735/213174966-2278afa2-5a02-4cbe-8cbe-5337f32ffa83.png)
NHWC
![image](https://user-images.githubusercontent.com/4530735/213174995-c6dd2939-ce57-48ae-a011-a8bbcb8bd027.png)
pytorch has problems with batchnorm support in NHWC format data and model.
